File: app/src/main/java/com/ianarbuckle/dublinbushelper/transports/schedules/ScheduleViewHolder.java
package com.ianarbuckle.dublinbushelper.transports.schedules;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.ianarbuckle.dublinbushelper.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Ian Arbuckle on 27/07/2017.
 *
 */

public class ScheduleViewHolder extends RecyclerView.ViewHolder implements ScheduleRowView {

  @BindView(R.id.tvDestination)
  TextView tvDest;

  @BindView(R.id.tvDue)
  TextView tvDue;

  @BindView(R.id.tvDirection)
  TextView tvDirection;

  public ScheduleViewHolder(View itemView) {
    super(itemView);
    ButterKnife.bind(this, itemView);
  }

  @Override
  public void setDestination(String destination) {
    tvDest.setText(destination);
  }

  @Override
  public void setDueTime(String dueTime) {
    tvDue.setText(dueTime);
  }

  @Override
  public void setDirection(String direction) {
    tvDirection.setText(direction);
  }
}

File: app/src/main/java/com/ianarbuckle/dublinbushelper/transports/schedules/ScheduleViewHolder.kt
package com.ianarbuckle.dublinbushelper.transports.schedules

import android.support.v7.widget.RecyclerView
import android.view.View

import kotlinx.android.synthetic.main.layout_item_schedule_data.view.*

/**
 * Created by Ian Arbuckle on 27/07/2017.

 */

class ScheduleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), ScheduleRowView {

    override fun setDestination(destination: String) {
        itemView.tvDest.text = destination
    }

    override fun setDueTime(dueTime: String) {
        itemView.tvDue.text = dueTime
    }

    override fun setDirection(direction: String) {
        itemView.tvDirection.text = direction
    }
}
